### PR TITLE
Nested sampling refactoring

### PR DESCRIPTION
**Pull request by ben18785**

### CODE REVIEW

File: pints/_nested/_rejection.py [hunk 1]
```diff
@@ -34,218 +32,29 @@ class NestedRejectionSampler(pints.NestedSampler):
...
+        self._proposed = self._log_prior.sample()[0]
+        return self._proposed
 
     def n_hyper_parameters(self):
-        """
-        Returns the number of hyper-parameters for this method (see
-        :class:`TunableMethod`).
-        """
+        """ See :meth:`TunableMethod.n_hyper_parameters()`. """
         return 1
```
Comment (MichaelClerx) [inline pints/_nested/_rejection.py]: Is this the same for all nested samplers, or should we be doing this on a per-sampler basis?

**Comment by MichaelClerx** [comment]
Looking good @ben18785 ! Let me know if I can help with this

**Comment by MichaelClerx** [comment]
Have fixed the mess I made of this branch with `git`. Please delete your local copy and re-checkout the branch before continuing work on this.

**Comment by MichaelClerx** [comment]
We should probably sit down for this one sometime soonish, @ben18785 ?

**Comment by ben18785** [comment]
Yes, I agree. Let's do it on Tuesday!

**Pull request closed without merging**